File: App/app/src/main/java/com/example/econo_misons/database/dao/categoryDAO.java
package com.example.econo_misons.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.econo_misons.database.models.Category;


import java.util.List;

//Data access object with all the SQL Query for the category functions
@Dao
public interface categoryDAO {
    @Insert
    long addCategory(Category cat);
    @Update
    void updateCategory(Category cat);
    @Delete
    void deleteCategory(Category cat);
    @Query("SELECT * FROM Category")
    LiveData<List<Category>> getAllCategories();
    @Query("SELECT * FROM Category WHERE ID = :catID")
    LiveData<List<Category>> getCategoryByID(int catID);

}

File: app/src/main/java/com/example/lafo_cheuse/database/dao/CategoryDao.kt
package com.example.lafo_cheuse.database.dao

import android.database.Cursor
import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.lafo_cheuse.models.Budget
import com.example.lafo_cheuse.models.Category
import kotlinx.coroutines.flow.Flow

@Dao
interface CategoryDao {
    @Query("SELECT * FROM Category")
    fun getCategories(): LiveData<List<Category>>?

    @Query("SELECT * FROM Category WHERE name = :categoryName AND emoji = :categoryEmoji")
    fun getCategory(categoryName : String, categoryEmoji : String): LiveData<List<Category>>?

    @Insert(onConflict = OnConflictStrategy.ABORT)
    fun createCategory(category: Category)

    @Update
    fun updateCategory(category: Category)

    @Query("DELETE FROM Category WHERE categoryId = :bId")
    fun deleteCategory(bId: Long) : Int

    @Query("DELETE FROM Category")
    fun deleteAllCategories() : Int
}